plugin/trino-raptor-legacy/src/test/java/io/trino/plugin/raptor/legacy/metadata/TestDatabaseShardManager.java

```diff
@@ -331,14 +322,12 @@ public void testReplaceShardUuids()
 
         // verify that conflicting updates are handled
         newShards = ImmutableList.of(shardInfo(UUID.randomUUID(), nodes.get(0)));
-        try {
-            transactionId = shardManager.beginTransaction();
-            shardManager.replaceShardUuids(transactionId, tableId, columns, replacedUuids, newShards, OptionalLong.of(0));
-            fail("expected exception");
-        }
-        catch (TrinoException e) {
-            assertEquals(e.getErrorCode(), TRANSACTION_CONFLICT.toErrorCode());
-        }
+        transactionId = shardManager.beginTransaction();
+        long finalTransactionId = transactionId;
+        List<ShardInfo> finalNewShards = newShards;
+        assertTrinoExceptionThrownBy(() -> shardManager.replaceShardUuids(finalTransactionId, tableId, columns, replacedUuids, finalNewShards, OptionalLong.of(0)))
+                .hasErrorCode(TRANSACTION_CONFLICT)
+                .hasMessage("Table was updated by a different transaction. Please retry the operation.");
     }
```

Comment:
I propose to put line 325 inside lambda as soon as it was in try block before. Is any exception will happen here, it will be inside assertTrinoExceptionThrownBy

Reply:
@MariyaR I've fixed the code as suggested.